### PR TITLE
feat: set oauth origin dynamically based on host header

### PR DESCRIPTION
**Pull request by dbarrosop**
Solves https://github.com/nhost/hasura-auth/issues/426

### CODE REVIEW

File: src/routes/oauth/utils.ts [hunk 1]
```diff
@@ -138,8 +138,7 @@ export const createGrantConfig = (): GrantConfig =>
     },
     {
       defaults: {
-        origin: ENV.AUTH_SERVER_URL,
-        prefix: OAUTH_ROUTE,
+        prefix: `${ENV.AUTH_API_PREFIX}${OAUTH_ROUTE}`,
```
Comment (onehassan) [inline src/routes/oauth/utils.ts:141]: How come we need both the the `AUTH_API_PREFIX` and `OAUTH_ROUTE` here?

File: src/routes/oauth/index.ts [hunk 2]
```diff
@@ -147,6 +147,13 @@ export const oauthProviders = Router()
    * Grant middleware: handle the oauth flow until the callback
    * @see {@link file://./config/index.ts}
    */
+  .use((req, res, next) => {
+    res.locals.grant = {dynamic: {
+        origin: `${req.protocol}://${req.headers.host}`},
+    };
+    next();
+
+  })
```
Comment (onehassan) [inline src/routes/oauth/index.ts:156]: nitpick: empty line

File: src/app.ts [hunk 3]
```diff
@@ -46,4 +47,8 @@ process.on('uncaughtException', (err, origin) => {
   process.exit(1);
 });
 
+app.use('/', (rwq, res) => {
+  return sendError(res,  'bad-request');
```
Comment (onehassan) [inline src/app.ts:51]:
```suggestion
app.use('/', (req, res) => {
  return sendError(res,  'bad-request');
```

File: src/routes/oauth/utils.ts (hunk 1 above)
Comment (dbarrosop) [reply src/routes/oauth/utils.ts:141]: You need to pass to grant the entire prefix where routes will be mounted. If we pass  `OAUTH_ROUTE` we are only passing part of the route.

**Review by onehassan — APPROVED** [review]
LGTM 🙌

**Comment by onehassan** [comment]
Can you add a changeset as well

**Pull request merged**